internal/olm/operator/internal/registry_pod.go

```diff
@@ -0,0 +1,145 @@
+	registryBundleImage string
+
+	ctx        context.Context
+	kubeclient kubernetes.Interface
+}
+
+// GetKubeClient creates a kubernetes clientset using kubeconfig
+func GetKubeClient() (client kubernetes.Interface) {
+	var kubeconfig *string
+	if home := homedir.HomeDir(); home != "" { // Check if machine has home directory
+		// Read kubeconfig flag. If not provided, use config file $HOME/.kube/config
+		kubeconfig = flag.String("kubeconfig", filepath.Join(home, ".kube", "config"), "(optional) absolute path to the kubeconfig file")
+	} else {
+		kubeconfig = flag.String("kubeconfig", "", "absolute path to the kubeconfig file")
+	}
+	flag.Parse()
+
+	// Build configuration from the config file
+	config, err := clientcmd.BuildConfigFromFlags("", *kubeconfig)
+	if err != nil {
+		panic(err)
+	}
+
+	// Create kubernetes clientset, which can be used to create, delete, patch, list kubernetes resources
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		panic(err)
+	}
+
+	return clientset
+}
+
+// CreateBundleRegistry returns a registry pod built from an index image
+func (rp *RegistryPod) CreateBundleRegistry() *core.Pod {
+	var pod *core.Pod
+
+	// TODO: Customizable fields can be moved to RegistryPod
+	podObject := &core.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "bundle-registry-pod",
+			Namespace: "default",
+			Labels: map[string]string{
+				"app": "default",
+			},
+		},
+		Spec: core.PodSpec{
+			Containers: []core.Container{
+				{
+					Name:    "registry-grpc",
+					Image:   rp.registryImage,
+					Command: []string{"/bin/sh"},
+					Args: []string{
+						"-c",
+						getContainerCmd(rp.registryDBPath, rp.registryBundleImage),
+					},
+					Ports: []core.ContainerPort{
+						{Name: "registry-port", ContainerPort: rp.registryPodGRPCPort},
+					},
+				},
+			},
+		},
+	}
+
+	// Check if Pod already exists
+	pod, err := rp.kubeclient.CoreV1().Pods(podObject.Namespace).Get(rp.ctx, podObject.Name, metav1.GetOptions{})
+	if err != nil && errors.IsNotFound(err) {
+		// Create pod in kubernetes cluster using the above pod definition and clientset
+		pod, err = rp.kubeclient.CoreV1().Pods(podObject.Namespace).Create(rp.ctx, podObject, metav1.CreateOptions{})
+		if err != nil {
+			panic(err)
+		}
+
+		// Poll and verify that pod status is running
+		podCheck := wait.ConditionFunc(func() (done bool, err error) {
+			var p *core.Pod
+			p, err = rp.kubeclient.CoreV1().Pods(podObject.Namespace).Get(rp.ctx, pod.Name, metav1.GetOptions{})
+			if err != nil {
+				return true, fmt.Errorf("error getting pod %s %w", pod.Name, err)
+			}
+			if p.Status.Phase == core.PodRunning {
+				return true, nil
+			}
+			return false, nil
+		})
+
+		err = wait.PollImmediateUntil(time.Duration(1*time.Second), podCheck, rp.ctx.Done())
+
+		if err != nil {
+			fmt.Printf("pod with the name %s hasn't been created successfully", podObject.Name)
+		}
+	}
+
+	return pod
+}
+
+func getContainerCmd(dbPath, bundlePath string) string {
```

Comment:
Make this a method on the `RegistryPod` struct, and use the struct fields to populate it?

```suggestion
func (r *RegistryPod) getContainerCmd() string {
```